aten/src/ATen/core/type.cpp

```diff
@@ -1410,14 +1410,48 @@ VaryingShape<Stride> TensorType::computeStrideProps(
   std::vector<size_t> stride_indices(sizes.size());
   std::iota(stride_indices.begin(), stride_indices.end(), 0);
 
+  // Sorting strides in ascending order
+  // Warning: A tensor that has more than one dimension of size 1 has
+  // insufficient information to recreate the contiguous order of its indices.
+  // Ties are broken based on whether one of the dimensions is of size
+  // one.  When two dimensions have the same stride, the stride
+  // associated with a dimension of size 1 is considered "smaller"
+  // as it created the condition for the second stride of the same size.
+  // Example:
+  //  Prior to sorting
+  //  Idx:     [0,   1,  2,  3]
+  //  sizes:   [8,   1, 10, 16]
+  //  Strides: [160, 1, 16,  1]
+  //  After sorting
+  //  Idx:     [1,  3,  2,   0]
+  //  sizes:   [1, 16, 10,   8]
+  //  Strides: [1,  1, 16, 160]
   std::sort(
       stride_indices.begin(),
       stride_indices.end(),
-      [&strides](const int& a, const int& b) {
-        // break ties in case of unsqueezed dims
-        // i.e. (1, 1, 5)
+      [&strides, &sizes](const int& a, const int& b) {
         if (strides[a] == strides[b]) {
-          return a > b;
+          // when strides of two axes are identical:
+          // i. if it's not self-overlapping:
+          //   1. if axis_a is the only size-1 axis, we think axis_a comes
+          //      before axis_b. Since strides[b] == strides[a] * sizes[a];
+          //      Meanwhile, it is not true for axis a;
+          //   2. if axis_b is the only size-1 axis, likewise, we think axis_a
+          //      comes after axis_b;
+          //   3. when both axes are size-1, we just assume canonical memory
+          //      format in PyTorch and takes whichever axis that comes first to
+          //      be the outer dimension;
+          //
+          // ii. when it's self-overlapping, we just default to PyTorch
+          //   canonical memory format and checks for whichever axis that comes
+          //   first;
+          if (sizes[a] == 1 && sizes[b] != 1) {
+            return true;
+          } else if (sizes[a] != 1 && sizes[b] == 1) {
+            return false;
+          } else {
+            return a > b;
+          }
         }
         return strides[a] < strides[b];
       });
```

Comment:
Look below at line 1469 https://github.com/pytorch/pytorch/pull/63940/files#diff-adcb9abb3299ad7384628ad8133a0ffa7f09b0491cdc2346af9c024cd492eab6R1469, looks like a long standing typo (should be `contiguous_ = sizes[stride_indices[i]] == 1`), amazing that it hasn't been caught.